src/oscar/apps/customer/utils.py

```diff
@@ -147,6 +147,6 @@ def normalise_email(email):
     """
     clean_email = email.strip()
     if '@' in clean_email:
-        local, host = clean_email.split('@')
+        local, host = clean_email.rsplit('@', 1)
```

Comment:
I changed the solution by removing normalize func from search form, but then i read this topic: https://stackoverflow.com/questions/2049502/what-characters-are-allowed-in-an-email-address
And according to it email ("very.unusual.@.unusual.com"@example.com) is allowed. So current normalize logic will fail on it with same error on registration or logic.
Also then i checked will validators pass emails with double "@" yes email validators allow it.
Also i django handle normalize same way as my first solution.

So currently im voting for my first solution.
Also i want to add this string to avoid None value exception.
`email = email or ''`

Reply:
I think you're right that `normalize_email` needs to be updated to support that unusual edge case. 

However I also don't think it is correct to use it in the search form. According to its docstring the search form is supposed to let you search for *partial addresses*, e.g., any email ending with `example.com`. We should not be expecting a fully valid email address, and so cannot expect to pass it through `normalize_email`.

So I would prefer to see solution 2 for the purpose of this issue, and then a separate change to `normalize_email` to support the edge case you've identified.

Reply:
Then current solution is fit for this issue. Soon will do a PR for normalize func.